src/components/PostDetails.tsx

```diff
@@ -1,118 +1,156 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { getComment, deleteComment } from '../api/data';
+
+type Props = {
+  activePostData: Post,
+};
+
+export const PostDetails: React.FC<Props> = React.memo(({ activePostData }) => {
+  const { id, title, body } = activePostData;
+
+  const [listOfComments, setListOfComments] = useState<null | Comment[]>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isGetCommentsError, setIsGetCommentsError] = useState(false);
+  const [showNewCommentForm, setShowNewCommentForm] = useState(false);
+
+  const [deletedCommentId, setDeletedCommentId] = useState<null | number>(null);
```

Comment:
Consider removing `deletedCommentId` state and filtering deleted comments directly in the `handleDeleteComment` function.

Reply:
Done